packages/taquito/src/prepare/prepare-provider.ts

```diff
@@ -1092,4 +1098,96 @@ export class PrepareProvider extends Provider implements PreparationProvider {
       contents,
     };
   }
+
+  /**
+   *
+   * @description Method to prepare a stake operation
+   * @param params RPCOperation object
+   * @param source string or undefined source pkh
+   * @returns a PreparedOperation object
+   */
+  stake(params: StakeParams): Promise<PreparedOperation> {
+    return this.staking({
+      ...params,
+      parameter: {
+        entrypoint: 'stake',
+        value: { prim: 'Unit' },
+      },
+    });
+  }
+
+  /**
+   *
+   * @description Method to prepare an unstake operation
+   * @param params RPCOperation object
+   * @param source string or undefined source pkh
+   * @returns a PreparedOperation object
+   */
+  async unstake({ amount, mutez, ...rest }: UnstakeParams): Promise<PreparedOperation> {
+    return this.staking({
+      parameter: {
+        entrypoint: 'unstake',
+        value: { int: mutez ? amount.toString() : format('tz', 'mutez', amount).toString() },
+      },
+      ...rest,
+      amount: 0,
+    });
+  }
+
+  /**
+   *
+   * @description Method to prepare a finalize_unstake operation
+   * @param params RPCOperation object
+   * @returns a PreparedOperation object
+   */
+  async finalizeUnstake(params: FinalizeUnstakeParams): Promise<PreparedOperation> {
+    return this.staking({
+      parameter: {
+        entrypoint: 'finalize_unstake',
+        value: { prim: 'Unit' },
+      },
+      ...params,
+      amount: 0,
```

Comment:
FinalizeUnstakeParams doesn't allow specify amount will go to `packages/taquito/src/prepare/prepare-provider.ts:235` to be '0'